src/Interpreters/TranslateQualifiedNamesVisitor.cpp

```diff
@@ -29,38 +31,56 @@ namespace ErrorCodes
     extern const int UNSUPPORTED_JOIN_KEYS;
     extern const int LOGICAL_ERROR;
 }
+bool TranslateQualifiedNamesMatcher::Data::matchColumnName(const String & name, const String & column_name, DataTypePtr column_type)
+{
+    if (name.size() < column_name.size())
+        return false;
+
+    if (std::strncmp(name.data(), column_name.data(), column_name.size()) != 0)
```

Comment:
here we can use start_with
```suggestion
    if (!name.starts_with(column_name))
```